### ISSUE TITLE
Sandboxed Google Play and Google Apps not completing MFA process with security keys

### ISSUE THREAD

**Issue by SirPanOfTheCake** [GrapheneOS/os-issue-tracker#2089]
I am having an issue with sand boxed play services and google apps (gmail, youtube and playstore). After signing in with username + password into the previously described Google apps, it never prompts to select option to use security key(NFC,USB..) See attached image for what screen it gets stuck on.

Using Vanidum with the same profile/device/Google account, I can sucessfully complete the authentication process to gmail and youtube with my security keys. I am also able use securitry keys in Bitwarden application in other profiles. 

I troubleshooted the issue with Tommy on the Grapheneos Off-Topic matrix channel. I have tried the following: 

- Give all permissons to Google Play Store and dependencies in App info -> issue remains 

- Created new profile, and install all sandboxed play services compents and repeated step above -> issue remains 

-Advised by GOS Matrix channel to give Google Play Store and dependencies services access to all files. Gave all permissons to Google Play Store and dependencies in App info , restarted device - > Issue remains 

Advised by Tommy to post issue and ping muhomorr on matrix.

**Pertinent Into**

Google Account used is enrolled into Google's Advanced Protection Program

**Device** 
Pixel 6

**Build** 
TQ2A.230305.008.E1.2023040400
![1](https://user-images.githubusercontent.com/104536736/230262116-7cc82069-023a-4d94-aab1-3f04b1e46eb8.JPG)

**Comment by muhomorr** [comment]
Need to see a logcat or a full bugreport: https://developer.android.com/studio/debug/bug-report#bugreportdevice

**Comment by SirPanOfTheCake** [comment]
@muhomorr I have DM'ed you the bug report on matrix.

**Comment by iwnc** [comment]
I have a similar issue. When I'm trying to authenticate using a security key in Google Play Store, after the NFC pairing it just sends me to the security key's website.

After aproximatelly 5 min an error appears, giving me the option to authenticate with security codes from a browser or to recover the account.  The codes don't work either, I tried about 5 of those 6 digit codes and everytime Google 2FA rejected them. When I choose to recover the account I got an email from Google with a link, opening the link starts Google Play Services but the app immediately crashes.

Let me know if you need more info.

Google Account used is enrolled into Google's Advanced Protection Program

Device
Pixel 6a

Build
TQ2A.230405.003.E1.2023041100

**Comment by SirPanOfTheCake** [comment]
Update, I have been troubleshooting with muhomorr.  I have tried updating Play services to current Alpha verison with no change to issue. 

It appears we were able to resolve the issue by doing the following:

- Unenroll my Google account from Google Advanced Protection Program
- Remove all hardware security keys as MFA options
- Enroll my Google account into Advanced Protection Program
- Registered the same security keys with Google Advanced Protection Program

Testing for the next few days to see if issue is resolved, but so far I have been able to log into multiple  Google accounts enrolled into Google Advanced Protection Program. I will post another update  this weekend.

**Comment by iwnc** [comment]
It didn't work for me. Still I'm getting redirected to security key's homepage (www.nitrokey.com).
What worked was generating a recovery code from a browser with the same IP address as my android device.

**Comment by jvoisin** [comment]
I was able to make the phone see my yubikey by explicitly enabling NFC devices in ` Settings > Connected Devices > Connection Preferences > NFC > Enable`.

Then I got the "You're all set, You can now remove your security key" message, but when I do, I get a "There was a problem. Try using your security key again or try another way to verify that it's you."

**Comment by Mello7sh3i** [comment]
> I was able to make the phone see my yubikey by explicitly enabling NFC devices in ` Settings > Connected Devices > Connection Preferences > NFC > Enable`.
> 
> Then I got the "You're all set, You can now remove your security key" message, but when I do, I get a "There was a problem. Try using your security key again or try another way to verify that it's you."
> 
> Trying like a dozen times, and then simply leaving the phone on top fo the yubikey for 10 minutes solved the issue.

https://issuetracker.google.com/issues/285558981
The problem is supposed to have been fixed a while ago, except if "it will become available in a future Android release" means it will be available for Android 26, since I just tried it, and it still doesn't work. No matter how long I wait, the error keeps looping back.
And unfortunately, NFC isn't only present on the Pixel tablet, so even if it were the solution, it wouldn't work everywhere.

**Comment by SirPanOfTheCake** [comment]
Posting a late update since I noticed others are having issues. I have multiple GOS devices (p6, p6a,p7,p7a and p8). This issue was intermittently coming back and affected seemilying random Google Acccounts with Advanced Protection enabled on multiple devices. I have noticed that this issue seemed to happen more with new Google accounts with advanced protection. 

Recently in the last month and last updates to GOS, I haven't been able to recreate the issue anymore. The issue appears be resolved for me.

Thanks @muhomorr for troubleshooting the issue with me and your work on the sandboxed play services.
